Stripe/UI/STPPaymentCardTextField.m

```diff
@@ -0,0 +1,704 @@
+        self.brandImageView.tintColor = placeholderColor;
+    }
+    
+    for (STPFormTextField *field in [self allFields]) {
+        field.placeholderColor = _placeholderColor;
+    }
+}
+
+- (UIColor *)placeholderColor {
+    return _placeholderColor ?: [self.class placeholderGrayColor];
+}
+
+- (void)setBorderColor:(UIColor * __nullable)borderColor {
+    self.layer.borderColor = [[borderColor copy] CGColor];
+}
+
+- (UIColor * __nullable)borderColor {
+    return [[UIColor alloc] initWithCGColor:self.layer.borderColor];
+}
+
+- (void)setCornerRadius:(CGFloat)cornerRadius {
+    self.layer.cornerRadius = cornerRadius;
+}
+
+- (CGFloat)cornerRadius {
+    return self.layer.cornerRadius;
+}
+
+- (void)setBorderWidth:(CGFloat)borderWidth {
+    self.layer.borderWidth = borderWidth;
+}
+
+- (CGFloat)borderWidth {
+    return self.layer.borderWidth;
+}
+
+- (void)setInputAccessoryView:(UIView *)inputAccessoryView {
+    _inputAccessoryView = inputAccessoryView;
+    
+    for (STPFormTextField *field in [self allFields]) {
+        field.inputAccessoryView = inputAccessoryView;
+    }
+}
+
+#pragma mark UIControl
+
+- (void)setEnabled:(BOOL)enabled {
+    [super setEnabled:enabled];
+    for (STPFormTextField *textField in [self allFields]) {
+        textField.enabled = enabled;
+    };
+}
+
+#pragma mark UIResponder & related methods
+
+- (BOOL)isFirstResponder {
+    return [self.selectedField isFirstResponder];
+}
+
+- (BOOL)canBecomeFirstResponder {
+    return [self.numberField canBecomeFirstResponder];
+}
+
+- (BOOL)becomeFirstResponder {
+    return [self.numberField becomeFirstResponder];
+}
+
+- (BOOL)canResignFirstResponder {
+    return [self.selectedField canResignFirstResponder];
+}
+
+- (BOOL)resignFirstResponder {
+    [super resignFirstResponder];
+    BOOL success = [self.selectedField resignFirstResponder];
+    [self setNumberFieldShrunk:[self shouldShrinkNumberField] animated:YES completion:nil];
+    return success;
+}
+
+- (BOOL)selectNextField {
+    return [[self nextField] becomeFirstResponder];
+}
+
+- (BOOL)selectPreviousField {
+    return [[self previousField] becomeFirstResponder];
+}
+
+- (STPFormTextField *)nextField {
+    if (self.selectedField == self.numberField) {
+        return self.expirationField;
+    } else if (self.selectedField == self.expirationField) {
+        return self.cvcField;
+    }
+    return nil;
+}
+
+- (STPFormTextField *)previousField {
+    if (self.selectedField == self.cvcField) {
+        return self.expirationField;
+    } else if (self.selectedField == self.expirationField) {
+        return self.numberField;
+    }
+    return nil;
+}
+
+#pragma mark public convenience methods
+
+- (void)clear {
+    for (STPFormTextField *field in [self allFields]) {
+        field.text = @"";
+    }
+    self.viewModel = [STPPaymentCardTextFieldViewModel new];
+    [self onChange];
+    [self updateImageForFieldType:STPCardFieldTypeNumber];
+    __weak id weakself = self;
```

Comment:
nit: the weak self dance isn't really necessary here, since nothing owns the completion block and it's fine for self to stick around until the animation completes